GraphActivity crashing

Here is adb logcat output
> 01-21 14:42:24.388 E AndroidRuntime: FATAL EXCEPTION: main
> 01-21 14:42:24.388 E AndroidRuntime: Process: cz.martykan.forecastie, PID: 16845
> 01-21 14:42:24.388 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{cz.martykan.forecastie/cz.martykan.forecastie.activities.GraphActivity}: java.lang.IllegalArgumentException: Number of rows/columns can't be lesser than 1.
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2951)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.os.Handler.dispatchMessage(Handler.java:106)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.os.Looper.loop(Looper.java:193)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.ActivityThread.main(ActivityThread.java:6718)
> 01-21 14:42:24.388 E AndroidRuntime:    at java.lang.reflect.Method.invoke(Native Method)
> 01-21 14:42:24.388 E AndroidRuntime:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
> 01-21 14:42:24.388 E AndroidRuntime:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
> 01-21 14:42:24.388 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Number of rows/columns can't be lesser than 1.
> 01-21 14:42:24.388 E AndroidRuntime:    at com.db.chart.view.ChartView.setGrid(ChartView.java:1423)
> 01-21 14:42:24.388 E AndroidRuntime:    at cz.martykan.forecastie.activities.GraphActivity.humidityGraph(GraphActivity.java:314)
> 01-21 14:42:24.388 E AndroidRuntime:    at cz.martykan.forecastie.activities.GraphActivity.onCreate(GraphActivity.java:107)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.Activity.performCreate(Activity.java:7144)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.Activity.performCreate(Activity.java:7135)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
> 01-21 14:42:24.388 E AndroidRuntime:    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
> 01-21 14:42:24.388 E AndroidRuntime:    ... 11 more

--- a/app/src/main/java/cz/martykan/forecastie/activities/GraphActivity.java
+++ b/app/src/main/java/cz/martykan/forecastie/activities/GraphActivity.java
@@ -130,7 +130,7 @@ public class GraphActivity extends BaseActivity {
         lineDataset.setThickness(4);
 
         int middle = Math.round(minTemp + (maxTemp - minTemp) / 2);
-        int stepSize = (int) Math.ceil(Math.abs(maxTemp - minTemp) / 4);
+        int stepSize = Math.max(1, (int) Math.ceil(Math.abs(maxTemp - minTemp) / 4));
         int min = middle - 2 * stepSize;
         int max = middle + 2 * stepSize;
 
@@ -212,14 +212,14 @@ public class GraphActivity extends BaseActivity {
         dataset.setThickness(4);
 
         int middle = Math.round(minPressure + (maxPressure - minPressure) / 2);
-        int stepSize = (int) Math.ceil(Math.abs(maxPressure - minPressure) / 4);
+        int stepSize = Math.max(1, (int) Math.ceil(Math.abs(maxPressure - minPressure) / 4));
         int min = middle - 2 * stepSize;
         int max = middle + 2 * stepSize;
         int rows = 4;
         if (Math.ceil(maxPressure) - Math.floor(minPressure) <= 3) {
             min = (int) Math.floor(minPressure);
-            max = (int) Math.ceil(maxPressure);
-            rows = (int) (Math.ceil(maxPressure) - Math.floor(minPressure));
+            max = Math.max(min + 1, (int) Math.ceil(maxPressure));
+            rows = max - min;
         }
 
         lineChartView.addData(dataset);
@@ -308,7 +308,11 @@ public class GraphActivity extends BaseActivity {
 
         int min = (int) minHumidity / 10 * 10;
         int max = (int) Math.ceil(maxHumidity / 10) * 10;
-        int stepSize = (max - min == 10) ? 20 : 10;
+        if (min == max) {
+            max = Math.min(max + 10, 100);
+            min = Math.max(min - 10, 0);
+        }
+        int stepSize = (max - min == 100) ? 20 : 10;
 
         lineChartView.addData(dataset);
         lineChartView.setGrid(ChartView.GridType.HORIZONTAL, (max - min) / stepSize, 1, gridPaint);
@@ -361,8 +365,6 @@ public class GraphActivity extends BaseActivity {
 
                 weather.setDate(listItem.getString("dt"));
                 weather.setTemperature(main.getString("temp"));
-
-                weather.setHumidity(main.getString("humidity"));
 
                 weatherList.add(weather);
             }

